crates/aptos/src/common/utils.rs

```diff
@@ -255,8 +255,29 @@ where
     Ok(map)
 }
 
-pub fn current_dir() -> PathBuf {
-    env::current_dir().unwrap()
+pub fn current_dir() -> CliTypedResult<PathBuf> {
+    env::current_dir().map_err(|err| {
+        CliError::UnexpectedError(format!("Failed to get current directory {}", err))
+    })
+}
+
+pub fn dir_default_to_current(maybe_dir: Option<PathBuf>) -> CliTypedResult<PathBuf> {
+    if let Some(dir) = maybe_dir {
+        Ok(dir)
+    } else {
+        current_dir()
+    }
+}
+
+pub fn create_dir_if_not_exist(dir: &Path) -> CliTypedResult<()> {
+    // Check if the directory exists, if it's not a dir, it will also fail here
```

Comment:
It wouldn't fail if the path exists but is not a dir right? Because the else case is hit.

Reply:
It is going to fail if it's not a dir:

```
$ touch test_file
$ ./target/debug/aptos move init --name name --package-dir test_file
{
  "Error": "Error accessing 'test_file/sources': Not a directory (os error 20)"
}

```